Source/RapyutaSimulationPlugins/Public/Core/RRConversionUtils.h

```diff
@@ -13,210 +13,179 @@
-                                 double& OutputX,
-                                 double& OutputY,
-                                 double& OutputZ)
-{
-    OutputX = InputX / 100.0f;
-    OutputY = -InputY / 100.0f;
-    OutputZ = InputZ / 100.0f;
-}
-
-inline static FVector RotationUEToROS(const FVector& Input)
-{
-    FVector Output = Input;
-
-    Output.Y = -Output.Y;
-    Output.Z = -Output.Z;
-
-    return Output;
-}
-
-inline static FQuat QuatUEToROS(const FQuat& Input)
-{
-    FQuat Output = Input;
-
-    Output.X = -Output.X;
-    Output.Z = -Output.Z;
-
-    return Output;
-}
-
-inline static FTransform TransformUEToROS(const FTransform& Input)
+UCLASS()
+class URRConversionUtils : public UBlueprintFunctionLibrary
 {
-    FTransform Output = Input;
+    GENERATED_BODY()
 
-    Output.SetTranslation(VectorUEToROS(Input.GetTranslation()));
-    Output.SetRotation(QuatUEToROS(Input.GetRotation()));
+public:
+    // UE to ROS conversion
+    // cm -> m
+    // Left handed -> Right handed
 
-    return Output;
-}
+    UFUNCTION(BlueprintCallable, Category = "Conversion")
+    static FVector ConvertHandedness(const FVector& InLocation)
+    {
+        return FVector(InLocation.X, -InLocation.Y, InLocation.Z);
+    }
 
-inline static FROSOdometry OdomUEToROS(const FROSOdometry& Input)
-{
-    FROSOdometry Output = Input;
-
-    VectorUEToROS(Input.pose_pose_position_x,
-                  Input.pose_pose_position_y,
-                  Input.pose_pose_position_z,
-                  Output.pose_pose_position_x,
-                  Output.pose_pose_position_y,
-                  Output.pose_pose_position_z);
-    Output.pose_pose_orientation = QuatUEToROS(Output.pose_pose_orientation);
-
-    Output.twist_twist_linear = VectorUEToROS(Output.twist_twist_linear);
-    Output.twist_twist_angular = RotationUEToROS(Output.twist_twist_angular);
-
-    return Output;
-}
-
-// // ROS to UE conversion
-// // m -> cm
-// // Right handed -> Left handed
-inline static FVector VectorROSToUE(const FVector& Input)
-{
-    FVector Output = Input;
+    UFUNCTION(BlueprintCallable, Category = "Conversion")
+    static FVector VectorUEToROS(const FVector& Input)
+    {
+        FVector Output = Input;
 
-    Output.X = Output.X * 100.0f;
-    Output.Y = -Output.Y * 100.0f;
-    Output.Z = Output.Z * 100.0f;
+        Output.X = Output.X;
+        Output.Y = -Output.Y;
+        Output.Z = Output.Z;
 
-    return Output;
-}
+        return 0.01f * Output;
+    }
 
-inline static void VectorROSToUE(const double& InputX,
-                                 const double& InputY,
-                                 const double& InputZ,
-                                 double& OutputX,
-                                 double& OutputY,
-                                 double& OutputZ)
-{
-    OutputX = InputX * 100.0f;
-    OutputY = -InputY * 100.0f;
-    OutputZ = InputZ * 100.0f;
-}
+    FORCEINLINE static void VectorUEToROS(const double& InputX,
```

Comment:
shouldn't add  FORCEINLINE for other methods as well?

Reply:
They are `BlueprintCallable` so serialized, thus not able to inline.